apps/mobile/src/store/settings/settings.ts

```diff
@@ -0,0 +1,128 @@
+import { useColorScheme } from 'react-native';
+import { useSelector } from 'react-redux';
+
+import { whenTheme } from '@/utils/when-theme';
+
+import {
+  AccountDisplayPreference,
+  BitcoinUnit,
+  DefaultNetworkConfigurations,
+  FiatCurrency,
+  WalletDefaultNetworkConfigurationIds,
+} from '@leather.io/models';
+
+import { useAppDispatch } from '../utils';
+import {
+  selectAccountDisplayPreference,
+  selectAnalyticsEnabled,
+  selectAppAuthEnabled,
+  selectBitcoinUnit,
+  selectConversionUnit,
+  selectNetwork,
+  selectTheme,
+  selectWalletSecurityLevel,
+} from './settings.read';
+import {
+  userChangedAccountDisplayPreference,
+  userChangedBitcoinUnit,
+  userChangedConversionUnit,
+  userChangedNetwork,
+  userChangedTheme,
+  userChangedWalletSecurityLevel,
+  userToggledAnalyticsEnabled,
+  userToggledAppAuthEnabled,
+} from './settings.write';
+
+export const defaultNetworks = ['mainnet', 'testnet', 'signet'] as const;
+
+export const defaultThemes = ['light', 'dark', 'system'] as const;
```

Comment:
Been thinking on this naming, because `system` isn't really a theme, it's a setting that points to a theme.

Maybe we have `ThemeSetting`/`ThemePreference` and `Theme`?

Reply:
Agree, been struggling a bit with some of the naming so thx for the feedback on all comments. I'll refactor them here.